internal/kubernetes/operator/datafederation/datafederation.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datafederation
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/mongodb/mongodb-atlas-cli/internal/kubernetes/operator/features"
+	"github.com/mongodb/mongodb-atlas-cli/internal/kubernetes/operator/resources"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	atlasV1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	DeletingState = "DELETING"
+	DeletedState  = "DELETED"
+)
+
+func BuildAtlasDataFederation(dataFederationStore store.DataFederationStore, dataFederationName, projectID, projectName, operatorVersion, targetNamespace string, dictionary map[string]string) (*atlasV1.AtlasDataFederation, error) {
+	dataFederation, err := dataFederationStore.DataFederation(projectID, dataFederationName)
```

Comment:
This method returns type mongodbatlas.DataFederationInstance. If store/data_federation.go will be changed to use [new SDK](https://github.com/mongodb/atlas-sdk-go), this method will return atlasv2.DataLakeTenant and all methods in this file will need to be adjusted accordingly.

Reply:
Done